public/store/AlertsStore.ts

```diff
@@ -16,49 +15,42 @@ export class AlertsStore {
 
   public async getAlertsByDetector(detectorId: string, detectorName: string) {
     let allAlerts: any[] = [];
-    const alertsSize = 10000;
-
-    const firstGetAlertsRes = await this.service.getAlerts({
-      detector_id: detectorId,
-      startIndex: 0,
-      size: alertsSize,
-    });
-
-    if (firstGetAlertsRes.ok) {
-      allAlerts = [...firstGetAlertsRes.response.alerts];
-      let remainingAlerts = firstGetAlertsRes.response.total_alerts - alertsSize;
-      let startIndex = alertsSize + 1;
-      const getAlertsPromises: Promise<ServerResponse<GetAlertsResponse>>[] = [];
+    const maxAlertsReturned = 10000;
+    let startIndex = 0;
+    let alertsCount = 0;
+
+    do {
+      const getAlertsRes = await this.service.getAlerts({
+        detector_id: detectorId,
+        startIndex,
+        size: maxAlertsReturned,
+      });
 
-      while (remainingAlerts > 0) {
-        getAlertsPromises.push(
-          this.service.getAlerts({ detector_id: detectorId, startIndex, size: alertsSize })
-        );
-        remainingAlerts -= alertsSize;
-        startIndex += alertsSize;
+      if (getAlertsRes.ok) {
+        allAlerts = allAlerts.concat(getAlertsRes.response.alerts);
+        alertsCount = getAlertsRes.response.alerts.length;
+      } else {
+        alertsCount = 0;
+        errorNotificationToast(this.notifications, 'retrieve', 'alerts', getAlertsRes.error);
       }
 
-      const alertsPromisesRes = await Promise.allSettled(getAlertsPromises);
+      startIndex += alertsCount;
+    } while (
+      // If we get 10,000 alerts as part of the previous call then there might be more alerts to fetch,
```

Comment:
Not blocking, but can we create an issue in the alerting/SAP backend repo to track fixing the getAlerts API response, and provide a link to that issue in this comment? That will help track when we should refactor this frontend code.

Reply:
@riysaxen-amzn Can you create an issue in backend and reference this PR there? Thanks!